Added reference to missing packages in README

AnimeshSinha1309 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
Fixes #15621 - But does not change the apt-get install to pip because I am not sure that is a good idea.

#### Brief description of what is fixed or changed
Made minor updates to the `README.rst` and `docs/README.rst` helping new users install dependencies when building docs

#### Other comments
* To `README.rst`
  - In the build docs section, added reference to docs/README.rst if package requirements are not satisfied.
* To `docs/README.rst`
  - Reduplicated the segment on installing python-mpmath if build fails in the Ubuntu/Debian segment.

#### Release Notes

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

AnimeshSinha1309 committed:
Added reference to missing packages in README

Fixing the issue #15621, added the refernce to `doc/README.rst` in the Make the Documentation segment of README, and added copied the instruction to install python3-mpmath in the Ubuntu subsection.

asmeurer commented on doc/README.rst:
```diff
@@ -16,6 +16,10 @@ and to view it, do::
 
+If you get **mpmath** error, install python3-mpmath package::
+
+    apt-get install python3-mpmath
```
I would just add this to the apt-get command above.

AnimeshSinha1309 commented:
@asmeurer I sure can add `python3-mpmath` or `python-mpmath` to the apt requirements in the install list. But I wanted to ask that should we not do this via **pip** instead of *apt-get*?

asmeurer commented:
I don't know what the general recommendation is for Linux whether to use pip or install the apt package? Someone who uses it will need to say. I guess pip also works even if you aren't using the system Python, which can also be a plus.

asmeurer commented:
Maybe the Python packages should be listed separately, noting you can install them with pip or with apt. Actually you also need sphinx, docutils, and matplotlib, see https://github.com/sympy/sympy/blob/master/.travis.yml#L251 (doctr is only required for the Travis build so ignore that).

AnimeshSinha1309 committed:
pip Packages in README

Changed README to have `pip install package` instead of `apt-get install python-package`.
